internal/go/skycfg/time.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2018 The Skycfg Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package skycfg
+
+import (
+	"time"
+
+	"go.starlark.net/starlark"
+)
+
+// TimeModule returns a new starlark.Value representing time Skycfg module.
+func TimeModule() starlark.Value {
+	return &Module{
+		Name: "time",
+		Attrs: starlark.StringDict{
+			"duration": starlark.NewBuiltin("time.duration", fnDuration),
+		},
+	}
+}
+
+func fnDuration(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var s string
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &s); err != nil {
+		return nil, err
+	}
+
+	d, err := time.ParseDuration(s)
```

Comment:
I'm not sure about exposing the Go duration syntax directly in the skycfg API, because it's difficult to implement in other versions of the Starlark language (Rust, Java).

Wouldn't `connect_timeout = Duration(seconds = 1)` be fine?

Reply:
> I'm not sure about exposing the Go duration syntax directly in the skycfg API, because it's difficult to implement in other versions of the Starlark language (Rust, Java).

Yeah, not sure either. Another option is to do implicit conversion to/from `google.protobuf.Duration` which seems very un-intuitive (to the user)...

> Wouldn't connect_timeout = Duration(seconds = 1) be fine, using the protobuf duration message directly?

This would require special-casing in `valueFromStarlark` to convert from `google.protobuf.Duration` (wrapped into `skyProtoMessage`) to plain `time.Duration`.

I can go ahead an change it if we are settled that implicit conversion is the way to go. lmk

Reply:
I'd like to hide the gogo-proto weirdness as much as possible. These fields _should_ be a `*duration_pb.Duration` in the generated Go, and it's only gogo's bizarro options that are turning it into some other type.